assets/js/googlesitekit-idea-hub-notice.js

```diff
@@ -47,8 +47,13 @@ const loadIdeaHubNotices = async ( _global = global ) => {
 	};
 
 	const listener = async () => {
+		// Added a no operation function as default value if the function does not exist on the store.
+		const noOperation = () => {};
+		const editorData = wp.data.select( 'core/editor' ) || {};
 		// eslint-disable-next-line sitekit/acronym-case
-		const postID = wp.data.select( 'core/editor' ).getCurrentPostId();
+		const { getCurrentPostId = noOperation } = editorData;
+		// eslint-disable-next-line sitekit/acronym-case
+		const postID = getCurrentPostId();
```

Comment:
This can be done a bit more straightforwardly, I think. The IB mentions checking to see if `wp.data.select( 'core/editor' ) === null`, which it will be when that store isn't loaded. We can return earlier here and not need to create/assign/run the no-op function, which I think is clearer logic to follow, eg:

```js
if ( wp.data.select( 'core/editor' ) === null ) {
	return;
}
```

Reply:
I do agree, wanted to ask / suggest going with the optional chaining operator instead so we don't introduce a new conditional.

```js
const postID = wp.data.select( 'core/editor' )?.getCurrentPostId();
```

Then the next (existing) conditional would return as `postID` would be `undefined`. Let me know what do you think about this approach.

Thanks for the review.

Reply:
Oh, yes, that's solid. We use optional chaining elsewhere so this makes total sense. Good catch! 👍🏻